internal/controllers/reconciliation/controller.go

```diff
@@ -177,7 +176,7 @@ func (c *Controller) reconcileResource(ctx context.Context, comp *apiv1.Composit
 		reconciliationLatency.Observe(time.Since(start).Seconds())
 	}()
 
-	if resource.Deleted() {
+	if resource.Deleted() || (resource.Patch != nil && resource.PatchSetsDeletionTimestamp()) {
```

Comment:
Does it make sense to move the rest of this condition to `Deleted`? Or move it and rename to `ShouldDelete`?

Reply:
Nice - fixed 👍